### ISSUE TITLE
get_predicted(): zero-inflation options

### ISSUE THREAD

**Issue by bwiernik** [easystats/insight#413]
Currently, for zero-inflated models, `get_predicted()` and its downstream functions like `modelbased::estimate_expectation()` always return the equivalent to `type = "conditional"` (predicted values assuming non-zero). It would be good to allow users to specify other methods, such as predicting unconditional response predictions (incorporating both parts of the model) or just the zero-inflation parts.

It would also be good I think to make the default equivalent to `type = "response"` (incorporating both model parts).

See the `type` argument in `predict.glmmTMB()`:

> type	
Denoting mu as the mean of the conditional distribution and p as the zero-inflation probability, the possible choices are:
>
>"link"
conditional mean on the scale of the link function, or equivalently the linear predictor of the conditional model
>"response"
expected value; this is mu*(1-p) for zero-inflated models and mu otherwise
>"conditional"
mean of the conditional response; mu for all models (i.e., synonymous with "response" in the absence of zero-inflation
>"zprob"
the probability of a structural zero (gives an error for non-zero-inflated models)
>"zlink"
predicted zero-inflation probability on the scale of the logit link functio>
>"disp"
dispersion parameter however it is defined for that particular family as described in sigma.glmmTMB

**Comment by DominiqueMakowski** [comment]
do you have a reproducible example with a model that has these components so I can play around?

**Comment by DominiqueMakowski** [comment]
``` r
library(glmmTMB)

m <- glmmTMB(count ~ spp + mined + (1|site),
             zi=~spp + mined,
             family=nbinom2, data=Salamanders)


head(insight::get_predicted(m))
#> [1] 0.5387752 1.0768783 0.3554236 2.4701755 2.4950498 2.1819828
head(insight::get_predicted(m, type = "zprob"))
#> [1] 2.040119 2.040119 2.040119 1.174339 1.174339 1.174339
```

<sup>Created on 2021-08-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

custom types should work after the latest commit. 

So now it becomes a question or wether we want to change / add to the behaviour of our main `predict` argument (this is also what would drive https://github.com/easystats/modelbased/issues/136). We could have `predict = "dispersion"` or something like that, but then again I'm not very familiar with these models so I don't know

**Comment by bwiernik** [comment]
Yes, I think the predict argument should have options for zero inflated and dispersion parameters

**Comment by bwiernik** [comment]
I think we should revert the `type` argument. Our `predict` argument fills the same role, and it's confusing to have two. 

Instead, I think we add options to `predict` to include these:

**Existing**
- "link"
    - linear predictor on the link scale (the conditional part for zero-inflated models)
    - with confidence intervals (uncertainty intervals on linear prediction)
    - same as glmmTMB's `type = "link"`

**Existing labels, need adjusted behavior**
- "expectation"
  - expected value (mean) on the response scale, including both the zero-inflated and conditional parts
  - with confidence intervals (uncertainty intervals on the conditional mean)
  - currently, it ignores the zero-inflated part
  - should be mu*(1-p) for zero-inflated models and mu otherwise
  - This would be equivalent to glmmTMB's `type = "response"`
- "prediction"
    - expected value (mean) on the response scale, including both the zero-inflated and conditional parts
    - with predction intervals (uncertainty intervals on the individual cases)
- "response"
  - "prediction", but classifying probabilities in binomial models to 0-1
  - uncertainty intervals would generally become [0, 1] for bernoulli models, but could be a range of integers for binomial models with multiple trials 

**New labels**
- "conditional"
  - expected value (mean) on the response scale, conditional on non-structural zero
  - with confidence intervals (uncertainty intervals on the conditional mean)
  - what is currently returned by `predict = "expectation"`
- "zprob"
  - expected probability of a structural zero
  - with confidence intervals (uncertainty intervals on the expected probability)
  - effectively the `"expectation"` for the zero-inflation part of the model
  - should give an error or message for non-zero-inflated models
- "zlink"
  - linear predictor for a structural zero on the link scale
  - with confidence intervals (uncertainty intervals on the linear prediction)
  - effectively the `"expectation"` for the zero-inflation part of the model
  - should give an error or message for non-zero-inflated models
- "dispersion"
  - expected value for the dispersion parameter for the model (e.g., conditional SD for a normal linear model)
  - with confidence intervals (uncertainty intervals on the conditional dispersion)
  - For gaussian models with `type = "disp"`, glmmTMB returns estimates/CI on the sigma (SD) scale; I don't see a need to offer other scales (variance or log-variance [this is what is actually modeled])

For `predict = "prediction"`, we should include the dispersion parameter in the prediction intervals. @DominiqueMakowski If you could add a placeholder for that, I can fill in the necessary extractions from glmmTMB objects. How do we currently handle dispersion for things like Poisson models where there is a variance term in the model?

**Comment by DominiqueMakowski** [comment]
The needed steps to add/edit I believe are:

- extending the list of possible arguments here:

https://github.com/easystats/insight/blob/1540c06176fb8c9994cc3348fc1eb0b09c971732/R/get_predicted.R#L275

- editing the logic here for the type:

https://github.com/easystats/insight/blob/1540c06176fb8c9994cc3348fc1eb0b09c971732/R/get_predicted.R#L645-L652

- and here for the CI

https://github.com/easystats/insight/blob/1540c06176fb8c9994cc3348fc1eb0b09c971732/R/get_predicted.R#L633-L643


(essentially since we have one "master" argument predict, it is then passed to the `.get_predicted_args()` helper that assigns the traditional arguments)

**Comment by strengejacke** [comment]
@DominiqueMakowski can _insight_ be submitted, or is there anything that needs to be addressed for _modelbased_?

**Comment by DominiqueMakowski** [comment]
I wouldn't say that this issue of adding more options for glmmTMB is urgent so probably not a blocker for a CRAN update

**Comment by strengejacke** [comment]
I _think_ this will be closed in #501

**Comment by strengejacke** [comment]
should be resolved in #501 to #503
